cmd/podman/kube/play.go

```diff
@@ -278,16 +286,58 @@ func play(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	if err := kubeplay(reader); err != nil {
-		// teardown any containers, pods, and volumes that might have created before we hit the error
-		teardownReader, trErr := readerFromArg(args[0])
-		if trErr != nil {
-			return trErr
+	// Create a channel to catch an interrupt or SIGTERM signal
+	ch := make(chan os.Signal, 1)
+	var teardownReader *bytes.Reader
+	if playOptions.Wait {
+		// Stop the the shutdown signal handler so we can actually clean up after a SIGTERM or interrupt
+		if err := shutdown.Stop(); err != nil {
+			return err
+		}
+		signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+		playOptions.ServiceContainer = true
+
+		// Read the kube yaml file again so that a reader can be passed down to the teardown function
+		teardownReader, err = readerFromArg(args[0])
+		if err != nil {
+			return err
 		}
-		if tErr := teardown(teardownReader, entities.PlayKubeDownOptions{Force: true}, true); tErr != nil && !errorhandling.Contains(tErr, define.ErrNoSuchPod) {
-			return fmt.Errorf("error tearing down workloads %q after kube play error %q", tErr, err)
+		fmt.Println("Use ctrl+c to clean up or wait for pods to exit")
+	}
+
+	cancelled := false
+	if playOptions.Wait {
+		// use a goroutine to wait for a sigterm or interrupt
+		go func() {
+			<-ch
+			// clean up any volumes that were created as well
+			fmt.Println("\nCleaning up containers, pods, and volumes...")
+			cancelled = true
+			if err := teardown(teardownReader, entities.PlayKubeDownOptions{Force: true}, false); err != nil && !errorhandling.Contains(err, define.ErrNoSuchPod) {
+				logrus.Errorf("Error during cleanup: %v", err)
+			}
+		}()
+	}
+
+	_, playErr := registry.ContainerEngine().PlayKube(registry.GetContext(), reader, playOptions.PlayKubeOptions)
```

Comment:
It looks like we're not using the report anymore and don't handle printing created objects anymore. I didn't try it out locally but I expect tests to break.

Reply:
fixed